### PR TITLE
Do not upload anything when tests fail on travis.

### PR DESCRIPTION
**Pull request by goderbauer**
This used to be the logic before I made the tests portable.

### CODE REVIEW

File: dev/bots/travis_upload.sh [hunk 1]
```diff
@@ -17,3 +17,5 @@ fi
 
 # generate the API docs, upload them
 ./dev/bots/docs.sh
+
+coveralls-lcov $LCOV_FILE
```
Comment (Hixie) [inline dev/bots/travis_upload.sh]: OMG! This is the source of our coveralls problems!

cc @abarth

Comment (Hixie) [inline dev/bots/travis_upload.sh]: no, wait, never mind. after_success wouldn't run if test.sh failed...

**Comment by Hixie** [comment]
LGTM

File: dev/bots/travis_upload.sh (hunk 1 above)
Comment (goderbauer) [inline dev/bots/travis_upload.sh]: With this change, Travis will now be RED if coverage upload failed. I thought, that might be helpful. (Travis is ignoring any failures in `after_success`).

**Pull request merged**